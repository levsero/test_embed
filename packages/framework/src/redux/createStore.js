import { createStore, compose, applyMiddleware } from 'redux'
import { createLogger } from 'redux-logger'
import thunk from 'redux-thunk'
import { inDebugMode } from '@zendesk/widget-shared-services'
import { trackAnalytics } from 'src/redux/middleware/analytics'
import { sendBlips } from 'src/redux/middleware/blip'
import onStateChangeFn from 'src/redux/middleware/onStateChange/onStateChange'
import onStateChangeWrapper from 'src/redux/middleware/onStateChange/wrapper'
import persist from 'src/redux/middleware/persist'
import preventLoops from 'src/redux/middleware/preventLoops'
import queueCalls from 'src/redux/middleware/queue'
import resetActiveEmbed from 'src/redux/middleware/resetActiveEmbed'
import throttle from 'src/redux/middleware/throttle'
import reducer from 'src/redux/modules/reducer'

const reduxConsoleLogger = (storeName) => {
  return createLogger({
    collapsed: true,
    titleFormatter: (action) => [`${storeName}`, `%c${String(action.type)}`, '%c'].join(' '),
  })
}

const reduxDevtoolsAvailable = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
const reduxDevtoolsComposer = (storeName) => {
  return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({ name: storeName })
}

export default function (storeName = 'web_widget', options = {}) {
  const middlewares = [
    thunk,
    preventLoops,
    throttle(options.throttleEvents, options.allowedActionsFn),
    onStateChangeWrapper(onStateChangeFn),
    sendBlips,
    onStateChangeWrapper(resetActiveEmbed),
    trackAnalytics,
    persist,
    onStateChangeWrapper(queueCalls),
  ]

  let composeEnhancers = compose

  if (inDebugMode()) {
    if (reduxDevtoolsAvailable) {
      composeEnhancers = reduxDevtoolsComposer(storeName)
    } else {
      middlewares.push(reduxConsoleLogger(storeName))
    }
  }

  return composeEnhancers(applyMiddleware(...middlewares))(createStore)(reducer)
}
