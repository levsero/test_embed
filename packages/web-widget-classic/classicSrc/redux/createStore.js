import { trackAnalytics } from 'classicSrc/redux/middleware/analytics'
import { sendBlips } from 'classicSrc/redux/middleware/blip'
import onStateChangeFn from 'classicSrc/redux/middleware/onStateChange/onStateChange'
import onStateChangeWrapper from 'classicSrc/redux/middleware/onStateChange/wrapper'
import persist from 'classicSrc/redux/middleware/persist'
import preventLoops from 'classicSrc/redux/middleware/preventLoops'
import queueCalls from 'classicSrc/redux/middleware/queue'
import resetActiveEmbed from 'classicSrc/redux/middleware/resetActiveEmbed'
import throttle from 'classicSrc/redux/middleware/throttle'
import reducer from 'classicSrc/redux/modules/reducer'
import { createStore, compose, applyMiddleware } from 'redux'
import { createLogger } from 'redux-logger'
import thunk from 'redux-thunk'
import { inDebugMode } from '@zendesk/widget-shared-services'

const reduxConsoleLogger = (storeName) => {
  return createLogger({
    collapsed: true,
    titleFormatter: (action) => [`${storeName}`, `%c${String(action.type)}`, '%c'].join(' '),
  })
}

const reduxDevtoolsAvailable = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
const reduxDevtoolsComposer = (storeName) => {
  return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({ name: storeName })
}

export default function (storeName = 'web_widget', options = {}) {
  const middlewares = [
    thunk,
    preventLoops,
    throttle(options.throttleEvents, options.allowedActionsFn),
    onStateChangeWrapper(onStateChangeFn),
    sendBlips,
    onStateChangeWrapper(resetActiveEmbed),
    trackAnalytics,
    persist,
    onStateChangeWrapper(queueCalls),
  ]

  let composeEnhancers = compose

  if (inDebugMode()) {
    if (reduxDevtoolsAvailable) {
      composeEnhancers = reduxDevtoolsComposer(storeName)
    } else {
      middlewares.push(reduxConsoleLogger(storeName))
    }
  }

  return composeEnhancers(applyMiddleware(...middlewares))(createStore)(reducer)
}
