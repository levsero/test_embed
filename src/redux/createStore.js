import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import onStateChange from 'redux-on-state-change';
import createLogger from 'redux-logger';

import { store } from 'service/persistence';
import { getEnvironment } from 'src/util/utils';

import reducer from 'src/redux/modules/reducer';
import onStateChangeFn from 'src/redux/middleware/onStateChange';

import { sendBlips } from 'src/redux/middleware/blip';

function loggerTitleFormatter(storeName) {
  return (action, time, took) => [`%c[${storeName}] ${String(action.type)}`, `%c@ ${time}`].join(' ');
}

export default function(storeName = 'web_widget') {
  const enableLogging = __DEV__ || getEnvironment() === 'staging' || store.get('debug');
  const logger = createLogger({ titleFormatter: loggerTitleFormatter(storeName) });
  const devToolsExtension = window.parent.__REDUX_DEVTOOLS_EXTENSION__
    && window.parent.__REDUX_DEVTOOLS_EXTENSION__({ name: storeName });
  const middlewares = [
    thunk,
    onStateChange(onStateChangeFn),
    sendBlips
  ];
  let storeEnhancers;

  if (enableLogging) {
    storeEnhancers = !!devToolsExtension
      ? [applyMiddleware(...middlewares), devToolsExtension]
      : [applyMiddleware(...middlewares, logger)];
  } else {
    storeEnhancers = [applyMiddleware(...middlewares)];
  }

  return compose(...storeEnhancers)(createStore)(reducer);
}
