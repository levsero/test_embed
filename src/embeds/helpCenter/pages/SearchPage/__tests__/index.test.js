import { Component as SearchPage } from '../index'
import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import createStore from 'src/redux/createStore'

const renderComponent = inProps => {
  const props = {
    title: 'pageTitle',
    isMobile: false,
    isContextualSearchPending: false,
    hideZendeskLogo: false,
    showNextButton: true,
    articles: [],
    ...inProps
  }

  return render(
    <Provider store={createStore()}>
      <SearchPage {...props} />
    </Provider>
  )
}

describe('render', () => {
  describe('when not isContextualSearchPending', () => {
    it('matches snapshot, renders NoResults', () => {
      expect(renderComponent().container).toMatchSnapshot()
    })
  })

  describe('when isContextualSearchPending', () => {
    it('matches snapshot, renders LoadingBarContent', () => {
      const result = renderComponent({ isContextualSearchPending: true })
      expect(result.container).toMatchSnapshot()
    })
  })

  it('renders button', () => {
    expect(renderComponent().getByText('Leave us a message')).toBeInTheDocument()
  })

  it('renders page title', () => {
    expect(renderComponent().getByText('pageTitle')).toBeInTheDocument()
  })

  describe('no results', () => {
    it('focuses on search field', () => {
      const { getByPlaceholderText } = renderComponent({ articles: [] })
      expect(document.activeElement).toEqual(getByPlaceholderText('How can we help?'))
    })
  })
})
