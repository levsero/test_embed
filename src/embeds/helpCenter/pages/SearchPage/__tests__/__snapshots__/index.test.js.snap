// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when isContextualSearchPending matches snapshot, renders LoadingBarContent 1`] = `
.c0 {
  border-radius: 0.5714285714285714rem 0.5714285714285714rem 0 0;
  padding: 0.5714285714285714rem 1rem;
  background: !important;
  color: !important;
}

.c3 {
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.02142857142857143rem;
  -moz-letter-spacing: 0.02142857142857143rem;
  -ms-letter-spacing: 0.02142857142857143rem;
  letter-spacing: 0.02142857142857143rem;
  line-height: 1.5;
  font-size: 1.0714285714285714rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 2px;
}

.c6:not(:first-child) {
  margin-top: 0.2857142857142857rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 2rem;
}

.c1:not(:first-child) {
  margin-top: 0.2857142857142857rem;
}

.c1 .c2:first-child {
  margin-left: 4rem;
}

.c1 .c2:nth-child(2) {
  margin-left: 2rem;
}

.c1 .c2:last-child {
  margin-right: 4rem;
}

.c1 .c2:nth-last-child(2) {
  margin-right: 2rem;
}

.c1 .c4 {
  width: 2rem !important;
  height: 2rem !important;
}

.c5 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 svg {
  width: 1rem !important;
  height: 1rem !important;
}

.c5:focus {
  outline: none !important;
  box-shadow: 0 0 0 0.21428571428571427rem rgba(255,255,255,0.4) !important;
}

.c13 {
  padding: 1.0714285714285714rem 1.4285714285714286rem;
  border-top: 0.07857142857142858rem solid #e9ebed;
  box-shadow: 0 -0.07142857142857142rem 0.8571428571428571rem rgba(0,0,0,0.08);
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c9 {
  color: #c2c8cc !important;
  height: 1.2rem !important;
  width: 1.2rem !important;
}

.c8 > div > div {
  max-height: unset;
}

.c7 {
  margin-bottom: 0;
}

.c15 {
  position: relative !important;
}

.c16 {
  padding-right: 0.7142857142857143rem !important;
  padding-left: 0.7142857142857143rem !important;
  max-width: 14.285714285714286rem;
}

.c10 {
  text-align: center !important;
  color: #2f3941;
  font-size: 1.0714285714285714rem;
  height: 80%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  margin-bottom: 0 !important;
  margin-top: 1.4285714285714286 !important;
}

.c12 {
  color: #68737d;
  margin-bottom: 1.4285714285714286 !important;
}

<div>
  <div
    class="container"
  >
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <h1
          class="c2 c3"
          data-testid="widget-title"
        >
          pageTitle
        </h1>
        <button
          aria-label="Minimize"
          class="c4 index__c-btn--icon___1Yl4x c5 index__c-btn___3y2Qe index__c-btn--basic___38OO7 index__c-btn--muted___ze6U0 index__c-btn--sm___3Kuhv "
          data-garden-container-id="keyboardfocus"
          data-garden-container-version="0.2.6"
          data-garden-id="buttons.icon_button"
          data-garden-version="7.0.2"
          tabindex="0"
          type="button"
        >
          <svg
            class="index__c-btn__icon___3V0Rl"
            data-garden-id="buttons.icon"
            data-garden-version="7.0.2"
            data-testid="Icon--dash"
            realfilename="dash-fill.svg"
          />
        </button>
      </div>
      <div
        class="c6"
      >
        <div
          class="form"
          data-testid="page-header-container"
        >
          <form
            class="c7"
            novalidate=""
          >
            <div
              class="c8"
            >
              <div
                class="text__c-txt___Uzv-G range__c-range___1aYaS "
              >
                <div
                  aria-invalid="false"
                  class="-div text__c-txt__input___38pRd text__c-txt__input--media___1dY36 text__is-focused___26DQb "
                  data-garden-id="forms.media_input"
                  data-garden-version="7.0.2"
                >
                  <div
                    class="text__c-txt__input--media__figure___gBDfx "
                  >
                    <svg
                      class="c9"
                      data-testid="Icon--search"
                      realfilename="search-stroke.svg"
                    />
                  </div>
                  <input
                    aria-invalid="false"
                    aria-labelledby="garden-field-container-2--label"
                    autocapitalize="off"
                    class="text__c-txt__input___38pRd text__c-txt__input--bare___wp3DV text__c-txt__input--media__body___11J0_ "
                    data-testid="search-field"
                    id="garden-field-container-2--input"
                    placeholder="How can we help?"
                    type="search"
                    value=""
                  />
                  <div
                    class="text__c-txt__input--media__figure___gBDfx "
                  >
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      class="container"
      data-testid="scroll-container-content"
    >
      <div
        class="c10"
      >
        <p
          class="c11"
        >
          There are no results for ""
        </p>
        <p
          class="c12"
        >
          Try searching for something else.
        </p>
      </div>
    </div>
    <footer
      class="c13"
    >
      <div
        class="c14"
      >
        <a
          class="icon"
          data-testid="Icon--zendesk"
          href="https://www.zendesk.com/embeddables/?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image"
          tabindex="-1"
          target="_blank"
        >
          <svg
            realfilename="widget-icon_zendesk.svg"
            title="zendesk"
          />
        </a>
        <div
          class="c15"
        >
          <div
            class="container buttonRight "
          >
            <button
              class="c16 index__c-btn___3y2Qe index__c-btn--primary___1YOfW "
              data-garden-container-id="keyboardfocus"
              data-garden-container-version="0.2.6"
              data-garden-id="buttons.button"
              data-garden-version="7.0.2"
              tabindex="0"
              type="button"
            >
              Leave us a message
            </button>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;

exports[`render when not isContextualSearchPending matches snapshot, renders NoResults 1`] = `
.c0 {
  border-radius: 0.5714285714285714rem 0.5714285714285714rem 0 0;
  padding: 0.5714285714285714rem 1rem;
  background: !important;
  color: !important;
}

.c3 {
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.02142857142857143rem;
  -moz-letter-spacing: 0.02142857142857143rem;
  -ms-letter-spacing: 0.02142857142857143rem;
  letter-spacing: 0.02142857142857143rem;
  line-height: 1.5;
  font-size: 1.0714285714285714rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 2px;
}

.c6:not(:first-child) {
  margin-top: 0.2857142857142857rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 2rem;
}

.c1:not(:first-child) {
  margin-top: 0.2857142857142857rem;
}

.c1 .c2:first-child {
  margin-left: 4rem;
}

.c1 .c2:nth-child(2) {
  margin-left: 2rem;
}

.c1 .c2:last-child {
  margin-right: 4rem;
}

.c1 .c2:nth-last-child(2) {
  margin-right: 2rem;
}

.c1 .c4 {
  width: 2rem !important;
  height: 2rem !important;
}

.c5 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 svg {
  width: 1rem !important;
  height: 1rem !important;
}

.c5:focus {
  outline: none !important;
  box-shadow: 0 0 0 0.21428571428571427rem rgba(255,255,255,0.4) !important;
}

.c13 {
  padding: 1.0714285714285714rem 1.4285714285714286rem;
  border-top: 0.07857142857142858rem solid #e9ebed;
  box-shadow: 0 -0.07142857142857142rem 0.8571428571428571rem rgba(0,0,0,0.08);
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c9 {
  color: #c2c8cc !important;
  height: 1.2rem !important;
  width: 1.2rem !important;
}

.c8 > div > div {
  max-height: unset;
}

.c7 {
  margin-bottom: 0;
}

.c15 {
  position: relative !important;
}

.c16 {
  padding-right: 0.7142857142857143rem !important;
  padding-left: 0.7142857142857143rem !important;
  max-width: 14.285714285714286rem;
}

.c10 {
  text-align: center !important;
  color: #2f3941;
  font-size: 1.0714285714285714rem;
  height: 80%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  margin-bottom: 0 !important;
  margin-top: 1.4285714285714286 !important;
}

.c12 {
  color: #68737d;
  margin-bottom: 1.4285714285714286 !important;
}

<div>
  <div
    class="container"
  >
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <h1
          class="c2 c3"
          data-testid="widget-title"
        >
          pageTitle
        </h1>
        <button
          aria-label="Minimize"
          class="c4 index__c-btn--icon___1Yl4x c5 index__c-btn___3y2Qe index__c-btn--basic___38OO7 index__c-btn--muted___ze6U0 index__c-btn--sm___3Kuhv "
          data-garden-container-id="keyboardfocus"
          data-garden-container-version="0.2.6"
          data-garden-id="buttons.icon_button"
          data-garden-version="7.0.2"
          tabindex="0"
          type="button"
        >
          <svg
            class="index__c-btn__icon___3V0Rl"
            data-garden-id="buttons.icon"
            data-garden-version="7.0.2"
            data-testid="Icon--dash"
            realfilename="dash-fill.svg"
          />
        </button>
      </div>
      <div
        class="c6"
      >
        <div
          class="form"
          data-testid="page-header-container"
        >
          <form
            class="c7"
            novalidate=""
          >
            <div
              class="c8"
            >
              <div
                class="text__c-txt___Uzv-G range__c-range___1aYaS "
              >
                <div
                  aria-invalid="false"
                  class="-div text__c-txt__input___38pRd text__c-txt__input--media___1dY36 text__is-focused___26DQb "
                  data-garden-id="forms.media_input"
                  data-garden-version="7.0.2"
                >
                  <div
                    class="text__c-txt__input--media__figure___gBDfx "
                  >
                    <svg
                      class="c9"
                      data-testid="Icon--search"
                      realfilename="search-stroke.svg"
                    />
                  </div>
                  <input
                    aria-invalid="false"
                    aria-labelledby="garden-field-container-0--label"
                    autocapitalize="off"
                    class="text__c-txt__input___38pRd text__c-txt__input--bare___wp3DV text__c-txt__input--media__body___11J0_ "
                    data-testid="search-field"
                    id="garden-field-container-0--input"
                    placeholder="How can we help?"
                    type="search"
                    value=""
                  />
                  <div
                    class="text__c-txt__input--media__figure___gBDfx "
                  >
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      class="container"
      data-testid="scroll-container-content"
    >
      <div
        class="c10"
      >
        <p
          class="c11"
        >
          There are no results for ""
        </p>
        <p
          class="c12"
        >
          Try searching for something else.
        </p>
      </div>
    </div>
    <footer
      class="c13"
    >
      <div
        class="c14"
      >
        <a
          class="icon"
          data-testid="Icon--zendesk"
          href="https://www.zendesk.com/embeddables/?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image"
          tabindex="-1"
          target="_blank"
        >
          <svg
            realfilename="widget-icon_zendesk.svg"
            title="zendesk"
          />
        </a>
        <div
          class="c15"
        >
          <div
            class="container buttonRight "
          >
            <button
              class="c16 index__c-btn___3y2Qe index__c-btn--primary___1YOfW "
              data-garden-container-id="keyboardfocus"
              data-garden-container-version="0.2.6"
              data-garden-id="buttons.button"
              data-garden-version="7.0.2"
              tabindex="0"
              type="button"
            >
              Leave us a message
            </button>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
